src/engine/srv_internal.h

```diff
@@ -326,17 +326,24 @@ dss_xs2tgt(int xs_id)
 		  "invalid xs_id %d, dss_tgt_nr %d, "
 		  "dss_tgt_offload_xs_nr %d.\n",
 		  xs_id, dss_tgt_nr, dss_tgt_offload_xs_nr);
+
+	if (xs_id < dss_sys_xs_nr)
+		return DSS_SYS_TGT_ID;
+
 	if (dss_helper_pool) {
-		if (xs_id < dss_sys_xs_nr ||
-		    xs_id >= dss_sys_xs_nr + dss_tgt_nr)
-			return -1;
+		if (xs_id >= dss_sys_xs_nr + dss_tgt_nr)
+			return DSS_INVALID_TGT_ID;
 		return xs_id - dss_sys_xs_nr;
 	}
 
-	if (xs_id < dss_sys_xs_nr)
-		return -1;
 	return (xs_id - dss_sys_xs_nr) /
 	       (dss_tgt_offload_xs_nr / dss_tgt_nr + 1);
 }
 
+static inline bool
+dss_xstream_has_context(struct dss_xstream *dx)
```

Comment:
dss_xstream_has_nvme()? Also, shouldn't we consider the tiers passed by control plane to avoid create nvme context on sys xstream when pmem tier is provided?

Reply:
Sure, will add a Todo here.